share_extension/ios/extension_teams.js

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {ActivityIndicator, FlatList, Text, View} from 'react-native';
+import DeviceInfo from 'react-native-device-info';
+import {intlShape} from 'react-intl';
+
+import {Client4} from 'mattermost-redux/client';
+import {General} from 'mattermost-redux/constants';
+
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+import ExtensionNavBar from './extension_nav_bar';
+import ExtensionTeamItem from './extension_team_item';
+
+export default class ExtensionTeams extends PureComponent {
+    static propTypes = {
+        currentTeamId: PropTypes.string.isRequired,
+        navigator: PropTypes.object.isRequired,
+        onSelectTeam: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired
+    };
+
+    static contextTypes = {
+        intl: intlShape
+    };
+
+    state = {
+        defaultChannels: null,
+        error: null,
+        myTeams: null
+    };
+
+    componentWillMount() {
+        this.loadTeams();
+    }
+
+    goBack = () => {
+        this.props.navigator.pop();
+    };
+
+    handleSelectTeam = (team) => {
+        const {defaultChannels} = this.state;
+        const townSquare = defaultChannels[team.id];
+        this.props.onSelectTeam(team, townSquare);
+        this.goBack();
+    };
+
+    keyExtractor = (item) => item.id;
+
+    loadTeams = async () => {
+        try {
+            const defaultChannels = {};
+            const teams = await Client4.getMyTeams();
+            const myMembers = await Client4.getMyTeamMembers();
+            const myTeams = teams.filter(async (team) => {
```

Comment:
Does using a promise in a filter work? That seems like a case where JavaScript would just interpret each promise as true and do no filtering